pkg/agent/route/route_linux.go

```diff
@@ -2994,3 +3180,206 @@ func (c *Client) deletePeerPodCIDRFromNFTablesSet(podCIDR *net.IPNet) error {
 
 	return nil
 }
+
+func (c *Client) getNFT(isIPv6 bool) knftables.Interface {
+	if isIPv6 {
+		return c.nftables.IPv6
+	}
+	return c.nftables.IPv4
+}
+
+func (c *Client) addNodePortConfigsNFTables(nodePortAddresses []net.IP, port uint16, protocol binding.Protocol) error {
+	isIPv6 := isIPv6Protocol(protocol)
+	transProtocol := getTransProtocolStr(protocol)
+	setName := getNodePortNFTablesSet(isIPv6)
+	nft := c.getNFT(isIPv6)
+
+	tx := nft.NewTransaction()
+	setElements := make([]*knftables.Element, 0, len(nodePortAddresses))
+	for i := range nodePortAddresses {
+		key := fmt.Sprintf("%s . %s . %d", nodePortAddresses[i], transProtocol, port)
+		elem := &knftables.Element{
+			Set: setName,
+			Key: []string{key},
+		}
+		tx.Add(elem)
+		setElements = append(setElements, elem)
+	}
+	if err := nft.Run(context.TODO(), tx); err != nil {
+		return fmt.Errorf("failed to add NodePort entries to nftables set %s: %w", setName, err)
+	}
+	c.serviceNFTablesSets[setName].Store(strconv.Itoa(int(port)), setElements)
```

Comment:
Won't the elements set be overwritten if there is a service with both TCP and UDP ports?